pkg/rancher-components/src/shim-vue.d.ts

```diff
@@ -1,11 +1,6 @@
-import { DefineComponent } from 'vue'
+export {}
 
-declare module '*.vue' {
-  const component: DefineComponent<{}, {}, any>
-  export default component
-}
-
-declare module '@vue/runtime-core' {
+declare module 'vue' {
```

Comment:
It appears that we want to provide type augments for `vue` rather than `@vue/runtime-core`[^1][^2]

[^1]: https://vuejs.org/guide/typescript/options-api#type-augmentation-placement
[^2]: https://github.com/lovetingyuan/router/blob/97be7c3462b1d44bf7767a76b7290b6764150895/packages/router/src/globalExtensions.ts#L11